File: Objective-C/Objective-C/SearchResultController.m
//
//  SearchResultController.m
//  Objective-C
//
//  Created by Nguyen Anh Minh on 2/19/16.
//  Copyright © 2016 Minh Nguyen. All rights reserved.
//

#import <CoreData/CoreData.h>

#import "SearchResultController.h"

NSString *const kCellIdentifier = @"TableViewCell";

@interface SearchResultController ()

@end

@implementation SearchResultController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.tableView registerNib:[UINib nibWithNibName:kCellIdentifier bundle:nil] forCellReuseIdentifier:kCellIdentifier];
}

- (void)setFilteredData:(NSArray *)filteredData{
    _filteredData = filteredData;
    [self.tableView reloadData];
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.filteredData.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = (UITableViewCell *)[self.tableView dequeueReusableCellWithIdentifier:kCellIdentifier];
    
    NSManagedObject *item = self.filteredData[indexPath.row];
    
    cell.textLabel.text = [item valueForKey:@"name"];
    
    return cell;
}

@end

File: SwiftSource/SwiftSource/SearchResultController.swift
//
//  SearchResultController.swift
//  SwiftSource
//
//  Created by Nguyen Anh Minh on 2/23/16.
//  Copyright © 2016 Minh Nguyen. All rights reserved.
//

import UIKit

class SearchResultController: UITableViewController {
    
    let cellIdentifier: String = "SearchResultCell"
    
    var filtedData: [Task]! = []{
        didSet{
            self.tableView.reloadData()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableView.delegate = self
        self.tableView.dataSource = self
        self.tableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: cellIdentifier)
    }
    
    // MARK: UITableView's Datasource & Delegate
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.filtedData.count
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier(cellIdentifier)
        
        let task = self.filtedData[indexPath.row]
        
        cell!.textLabel!.text = task.name
        
        return cell!
    }
}